ui/src/app/workflow-templates/components/workflow-template-filters/workflow-template-filters.tsx

```diff
@@ -51,7 +53,17 @@ export function WorkflowTemplateFilters({templates, namespace, labels, onChange}
                         autocomplete={labelSuggestion}
                         tags={labels}
                         onChange={tags => {
-                            onChange(namespace, tags);
+                            onChange(namespace, namePattern, tags);
+                        }}
+                    />
+                </div>
+                <div className='columns small-2 xlarge-12'>
+                    <p className='wf-filters-container__title'>Name Pattern</p>
```

Comment:
"Pattern" is a bit ambiguous as it can mean several things. "Contains" is a more explicit naming. See https://github.com/argoproj/argo-workflows/pull/13160#discussion_r1633477891 for more details